api/modules/files/cache.js

```diff
@@ -4,40 +4,73 @@ const Files = require(`./model`);
 
 const BaseCache = require(`../../classes/base_cache`);
 
+const DEFAULT_BUFFER_CACHE_EXPIRY_SEC = 10 * 60;
+
+function getBuffer(fileId, next) {
+  return Files.query({
+    where: {
+      id: fileId
+    },
+    columns: [`buffer`]
+  })
+  .fetch()
+  .then(file => next(null, file && file.get(`buffer`)))
+  .catch(next);
+}
+
 class RemixedFileCreationCache extends BaseCache {
-  static get name() {
-    return `createdRemixFile`;
+  constructor(cache) {
+    super(cache);
   }
 
-  static get config() {
-    return Object.assign(super.config, {
-      segment: `file_buffers`
-    });
+  get name() {
+    return `createdRemixFile`;
   }
 
-  // eslint-disable-next-line no-unused-vars
-  static generateKey(fileId, fileBuffer) {
-    return fileId;
+  get config() {
+    // We want to override the base config since we want to implement our own
+    // caching process
+    return null;
   }
 
-  static run(fileId, fileBuffer, next) {
+  /*
+   * This method takes an optional `fileBuffer` argument
+   * If the `fileBuffer` was passed in, it writes it to cache
+   * If not, the callback is called with a `fileBuffer` retrieved from cache
+   */
+  run(fileId, fileBuffer, next) {
+    /*
+     * Putting the file buffer into cache
+     */
+
     if(typeof fileBuffer !== `function`) {
-      return next(null, fileBuffer);
+      if(!this.cache) {
+        return next(null, fileBuffer);
+      }
+
+      return this.cache.setex(`file:${fileId}`, DEFAULT_BUFFER_CACHE_EXPIRY_SEC, fileBuffer)
```

Comment:
What about needing to inform redis underneath that we're using `Buffer` types as values?  Does it matter, or is it happening for free in the redis client catbox is using?